crates/db_schema/src/impls/actor_language.rs

```diff
@@ -90,24 +83,33 @@ impl LocalUserLanguage {
       .build_transaction()
       .run(|conn| {
         Box::pin(async move {
-          use crate::schema::local_user_language::dsl::{local_user_id, local_user_language};
-          // Clear the current user languages
-          delete(local_user_language.filter(local_user_id.eq(for_local_user_id)))
-            .execute(conn)
-            .await?;
+          use crate::schema::local_user_language::dsl::{
+            language_id,
+            local_user_id,
+            local_user_language,
+          };
+          // Delete old languages, not including new languages
+          let delete_old = delete(local_user_language)
+            .filter(local_user_id.eq(for_local_user_id))
+            .filter(language_id.ne_all(&lang_ids))
+            .execute(conn);
 
           let forms = lang_ids
-            .into_iter()
+            .iter()
```

Comment:
Looks like these changes break the tests, and anyway that was fine.

Reply:
I now tried to fix the type mismatch error.

This change is needed to satisfy the borrow checker.